05.wc/wc.rb

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options, file_paths = parse_options(argv: ARGV)
+
+  # パイプで接続した場合 $stdin.tty? は false になる
+  if file_paths.empty? && !$stdin.tty?
+    count_stats = count_content(content: $stdin.read)
+    max_width = calculate_max_width(file_info_list: [{ stats: count_stats }], total_stats: nil)
+
+    # file_pathにnilを渡すことで、ファイル名を出力しない
+    output_format(count_stats:, options:, file_path: nil, max_width:)
+    return
```

Comment:
```suggestion
 else
   # ...
```

早期リターンは「特定の条件に合致したらここでおしまい！」という状況を表現する場合は向いていますが、このコードの場合はあくまで「Aの場合はこう、それ以外ならこう」という関係にあると思うので、if/elseの方が読みやすいと思います。

Reply:
ちょっと安易にreturnを使う癖がついていたみたいです。
エラーハンドリング系は早期returnして、それ以外の条件分岐は if/elseになるようにしました。